src/libraries/System.Private.CoreLib/src/System/Collections/ListDictionaryInternal.cs

```diff
@@ -202,6 +203,17 @@ public void Remove(object key)
             count--;
         }
 
+        internal KeyValuePairs[] ToKeyValuePairsArray()
+        {
+            var array = new KeyValuePairs[count];
```

Comment:
Is it possible the list can change while executing this? I am asking if we can instead use `List<KVP>`?

Reply:
This code is executed only when the program is suspended by a debugger. Hashtable has similar code.

Reply:
ok, thanks!